src/main/java/uk/ac/ebi/subs/fileupload/services/globus/GlobusService.java

```diff
@@ -0,0 +1,359 @@
+                Files.createDirectories(targetPath);
+                LOGGER.debug("Setting permissions on upload directory for owner : {}, directory : {}", gs.getOwner(), targetPath.toString());
+
+                ProcessBuilder pb = new ProcessBuilder("/usr/bin/chmod", "g+w", targetPath.toString());
+                Process proc = pb.start();
+                if (proc.waitFor() != 0) {
+                    throw new RuntimeException("Error setting permissions on upload directory. Owner : " + gs.getOwner() + ", Directory : " + targetPath.toString());
+                }
+            } catch (Exception ex) {
+                LOGGER.debug("Error creating upload directory for owner : {}. Deleting share document.", gs.getOwner());
+
+                //delete the share object to make retry possible.
+                globusShareRepository.delete(gs.getId());
+
+                throw new RuntimeException("Error creating upload directory for owner : " + gs.getOwner(), ex);
+            }
+        } else {
+            LOGGER.debug("Upload directory already exists for owner : {}", gs.getOwner());
+        }
+    }
+
+    private GlobusShare createShareLink(GlobusShare gs) {
+        LOGGER.debug("Creating share link for owner : {}", gs.getOwner());
+
+        String sharedEndpointId = null;
+        try {
+            sharedEndpointId = globusApiClient.createShare(
+                    hostEndpointBaseDir + "/" + gs.getOwner(), UUID.randomUUID().toString(), "");
+        } catch (Exception ex) {
+            LOGGER.debug("Error creating globus share for owner : {}. Deleting share document.", gs.getOwner());
+
+            //delete the share object to make retry possible.
+            globusShareRepository.delete(gs.getId());
+
+            throw new RuntimeException("Error creating globus share for owner : " + gs.getOwner(), ex);
+        }
+
+        gs.setSharedEndpointId(sharedEndpointId);
+        gs.setShareLink(String.format(shareUrlFormat, sharedEndpointId));
+
+        LOGGER.debug("Updating share document with new endpoint and share link. Owner : {}", gs.getOwner());
+
+        return globusShareRepository.save(gs);
+    }
+
+    private File createFileObject(String owner, String submissionId, java.io.File file) {
+        Submission submission = submissionRepository.findOne(submissionId);
+
+        File fileObj = new File();
+        fileObj.setCreatedBy(submission.getSubmitter().getName());
+        fileObj.setFilename(file.getName());
+        fileObj.setGeneratedTusId(UUID.randomUUID().toString());
+        fileObj.setId(fileObj.getGeneratedTusId());
+        fileObj.setSource(FileSource.GLOBUS.toString());
+        fileObj.setStatus(FileStatus.UPLOADED);
+        fileObj.setSubmissionId(submissionId);
+        fileObj.setTotalSize(file.length());
+        fileObj.setUploadedSize(file.length());
+        fileObj.setUploadStartDate(LocalDateTime.now());
+        fileObj.setUploadFinishDate(fileObj.getUploadStartDate());

```

Comment:
in globus's case there is no way to tell when was the uploaded actually started. so i just used finish date as the start date here.

Reply:
Well, I think we would be able to set an approximate start date from the UI if the file upload would start with a click of a button that would take to user to the Globus UI. Although it would not be an exact date. That is another bad point for using Globus